plugins/invisible-typing/src/index.ts

```diff
@@ -12,7 +12,7 @@ export interface SettingsType {
   button?: boolean;
   invisible?: boolean;
   channelWise?: boolean;
-  channels?: Record<string, boolean>;
+  channels?: Record<string, boolean | undefined>;
```

Comment:
<a href="https://sourcery.ai"><img src="https://sourcery.ai/favicon.svg" height="15" alt="Sourcery logo" /></a>  **suggestion (code_refinement):** Adjustment to the 'channels' type to allow undefined values aligns with the nullish coalescing operator usage.

This type change is necessary to explicitly allow 'undefined' as a value for channel settings, which complements the logic introduced in the Icon component. It ensures type safety across the plugin.

```suggestion
channels?: Record<string, boolean | undefined>;
```